Part 2 - Landing Page/js/app.js

```diff
@@ -17,15 +17,22 @@
  * Comments should be present at the beginning of each procedure and class.
  * Great to have comments before crucial code sections within the procedure.
 */
+// Define Global Variables
+const ul = document.querySelector('#navbar__list');
+const sect1 = document.querySelector('#section1');
+const sect2 = document.querySelector('#section2');
+const sect3 = document.querySelector('#section3');
 
-/**
- * Define Global Variables
- * 
-*/
-const navigation = document.getElementById('navbar__list');
-const sections = document.querySelectorAll('section');
+// build navigation links
+const link1 = `<li><a class="menu__link" href="#${sect1.id}">${sect1.dataset.nav}</a></li>`;
+const link2 = `<li><a class="menu__link" href="#${sect2.id}">${sect2.dataset.nav}</a></li>`;
+const link3 = `<li><a class="menu__link" href="#${sect3.id}">${sect3.dataset.nav}</a></li>`;
```

Comment:
- I told you not to get data from page content.
- This is so much redundant code.

You should do the following instead:
const xxx = [ { id: xxx, text: xxx }, { id: xxx, text: xxx}, { id: xxx, text: xxx} ]
const linkArr = xxx.map(xxx => xxx)

Reply:
Done.

```
// build nav links
const links = [
    { id: '#section1', text: 'Section 1'},
    { id: '#section2', text: 'Section 2'},
    { id: '#section3', text: 'Section 3'},   
];

ul.innerHTML = links.map(link => link = `<li><a class="menu__link" href=${link.id}>${link.text}</a></li>`); 
```